client/x/evmstaking/keeper/keeper_test.go

```diff
@@ -127,7 +135,305 @@ func (s *TestSuite) SetupTest() {
+					createCorruptedPubKey(delPubKey.Bytes()),
+					"moniker",
+					new(big.Int).SetUint64(100),
+					uint32(1000),
+					uint32(5000),
+					uint32(500),
+				)
+				require.NoError(err)
+				logs := []ethtypes.Log{{Topics: []common.Hash{types.CreateValidatorEvent.ID}, Data: data}}
+				evmEvents, err := ethLogsToEvmEvents(logs)
+				if err != nil {
+					return nil, err
+				}
+
+				return evmEvents, nil
+			},
+		},
+		{
+			name: "pass(continue): fail to process DepositEvent - corrupted delegator pubkey",
+			evmEvents: func() ([]*evmenginetypes.EVMEvent, error) {
+				data, err := stakingAbi.Events["Deposit"].Inputs.NonIndexed().Pack(
+					uncompressedDelPubKeyBytes,
+					createCorruptedPubKey(delPubKey.Bytes()),
+					valPubKey.Bytes(),
+					new(big.Int).SetUint64(100),
+				)
+				require.NoError(err)
+				logs := []ethtypes.Log{{Topics: []common.Hash{types.DepositEvent.ID}, Data: data}}
+				evmEvents, err := ethLogsToEvmEvents(logs)
+				if err != nil {
+					return nil, err
+				}
+
+				return evmEvents, nil
+			},
+		},
+		{
+			name: "pass(continue): fail to process RedelegateEvent - corrupted delegator pubkey",
+			evmEvents: func() ([]*evmenginetypes.EVMEvent, error) {
+				data, err := stakingAbi.Events["Redelegate"].Inputs.NonIndexed().Pack(
+					createCorruptedPubKey(delPubKey.Bytes()),
+					valPubKey.Bytes(),
+					valPubKey2.Bytes(),
+					new(big.Int).SetUint64(100),
+				)
+				require.NoError(err)
+				logs := []ethtypes.Log{{Topics: []common.Hash{types.RedelegateEvent.ID}, Data: data}}
+				evmEvents, err := ethLogsToEvmEvents(logs)
+				if err != nil {
+					return nil, err
+				}
+
+				return evmEvents, nil
+			},
+		},
+		{
+			name: "pass(continue): fail to process WithdrawEvent - corrupted delegator pubkey",
+			evmEvents: func() ([]*evmenginetypes.EVMEvent, error) {
+				data, err := stakingAbi.Events["Withdraw"].Inputs.NonIndexed().Pack(
+					createCorruptedPubKey(delPubKey.Bytes()),
+					valPubKey.Bytes(),
+					new(big.Int).SetUint64(100),
+				)
+				require.NoError(err)
+				logs := []ethtypes.Log{{Topics: []common.Hash{types.WithdrawEvent.ID}, Data: data}}
+				evmEvents, err := ethLogsToEvmEvents(logs)
+				if err != nil {
+					return nil, err
+				}
+
+				return evmEvents, nil
+			},
+		},
+		{
+			name: "pass(continue): fail to process UnjailEvent - invalid validator pubkey",
+			evmEvents: func() ([]*evmenginetypes.EVMEvent, error) {
+				invalidValPubKey := valPubKey.Bytes()[1:]
+				data, err := slashingAbi.Events["Unjail"].Inputs.NonIndexed().Pack(invalidValPubKey)
+				require.NoError(err)
+				logs := []ethtypes.Log{{Topics: []common.Hash{types.UnjailEvent.ID, common.BytesToHash(delEvmAddr.Bytes())}, Data: data}}
+				evmEvents, err := ethLogsToEvmEvents(logs)
+				if err != nil {
+					return nil, err
+				}
+
+				return evmEvents, nil
+			},
+		},
+		// SUCCESS Cases should be tested separately with unit tests in the files where each processing logic is defined.
```

Comment:
Should include success cases here to make sure certain event will route to corresponding processing functions. One test for each routing should be good. Unit tests in those files do not test the routing logic.

Reply:
@limengformal 
I added success cases [here](https://github.com/piplabs/story/pull/75/files#diff-be42d583c30357b6b5b2cb6032c7cdbfdfcb98294609d73e099aeab38239d57dR410-R607)